### PR TITLE
Removed dependency on Azure Storage Common.

### PR DESCRIPTION
**Pull request by vcolin7**
Brought over all the Storage Common classes Tables relies on.

### CODE REVIEW

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java [hunk 1]
```diff
@@ -176,8 +168,8 @@ private static String getBearerTokenScope(UrlBuilder endpoint) {
      */
     private static HttpPipelinePolicy getResponseValidationPolicy() {
         return new ResponseValidationPolicyBuilder()
-            .addOptionalEcho(Constants.HeaderConstants.CLIENT_REQUEST_ID)
-            .addOptionalEcho(Constants.HeaderConstants.ENCRYPTION_KEY_SHA256)
+            .addOptionalEcho(StorageConstants.HeaderConstants.CLIENT_REQUEST_ID)
+            .addOptionalEcho(StorageConstants.HeaderConstants.ENCRYPTION_KEY_SHA256)
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java:172]: Is it required to validate that the response headers are the same as the request headers?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * Fluently applies an optional validation to this policy where, if the response contains the given header, asserts
+     * its value is an echo of the value provided in the request.
+     *
+     * @param headerName The header to validate.
+     *
+     * @return The updated {@link HttpPipelinePolicy}.
+     */
+    public ResponseValidationPolicyBuilder addOptionalEcho(String headerName) {
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java:40]: Just curious if we really need this validation.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java [hunk 3]
```diff
@@ -356,18 +356,18 @@ public TableClientBuilder serviceVersion(TablesServiceVersion version) {
...
      *
-     * @throws NullPointerException If {@code retryOptions} is {@code null}.
+     * @throws NullPointerException If {@code retryPolicy} is {@code null}.
      */
-    public TableClientBuilder retryOptions(RequestRetryOptions retryOptions) {
-        if (retryOptions == null) {
-            throw logger.logExceptionAsError(new NullPointerException("'retryOptions' cannot be null."));
+    public TableClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        if (retryPolicy == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'retryPolicy' cannot be null."));
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java:367]: (Re)setting `retryPolicy`  to `null`  should be allowed.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java [hunk 4]
```diff
@@ -351,18 +351,18 @@ public TableServiceClientBuilder serviceVersion(TablesServiceVersion version) {
...
      *
-     * @throws NullPointerException if {@code retryOptions} is {@code null}.
+     * @throws NullPointerException if {@code retryPolicy} is {@code null}.
      */
-    public TableServiceClientBuilder retryOptions(RequestRetryOptions retryOptions) {
-        if (retryOptions == null) {
-            throw logger.logExceptionAsError(new NullPointerException("'retryOptions' cannot be null."));
+    public TableServiceClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        if (retryPolicy == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'retryPolicy' cannot be null."));
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java:362]: `retryPolicy` can be set to null in which case we'll fall back to default retry policy.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableStorageException.java [hunk 5]
```diff
@@ -5,7 +5,8 @@
 
 import com.azure.core.exception.HttpResponseException;
 import com.azure.core.http.HttpResponse;
-import static com.azure.storage.common.implementation.Constants.HeaderConstants.ERROR_CODE;
+
+import static com.azure.data.tables.implementation.StorageConstants.HeaderConstants.ERROR_CODE;
 
 /**
  * A {@code TableStorageException} is thrown whenever the Tables service successfully returns an error code that is not
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableStorageException.java:12]: Isn't `TableStorageException` deleted now?

File: sdk/tables/azure-data-tables/pom.xml [hunk 6]
```diff
@@ -99,7 +99,6 @@ Licensed under the MIT License.
             <version>3.0.0-M3</version> <!-- {x-version-update;org.apache.maven.plugins:maven-surefire-plugin;external_dependency} -->
             <configuration>
               <argLine>
-                --add-opens com.azure.storage.common/com.azure.storage.common.implementation=ALL-UNNAMED
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/pom.xml:102]: Line numbers 45-49 should be deleted as well.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.function.BiConsumer;
+
+/**
+ * Builder for a policy to do validation of general response behavior.
+ */
+public class ResponseValidationPolicyBuilder {
```
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java:20]: Spotbugs complains about the policy not being in the public package, should I add an exception and move it back to `implementation`?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableStorageException.java (hunk 5 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableStorageException.java:12]: That work is part of a different PR.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java (hunk 1 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java:172]: I'm not sure, this was added before my time in Tables. I suspect it's copying what's done in the Storage libraries where this is validated. I can check with the rest of the team to see if other languages are doing the same validation.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java (hunk 2 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java:40]: See my comment above.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java (hunk 4 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java:362]: We'll do this in #21368.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableStorageException.java (hunk 5 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableStorageException.java:12]: #21339

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java (hunk 1 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java:172]: It seems like what I said above was the case. The other languages are not doing this at all, so I think we can remove it for the time being.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java:20]: Will remove this.

File: sdk/tables/azure-data-tables/pom.xml [hunk 8]
```diff
@@ -44,8 +44,8 @@ Licensed under the MIT License.
     </dependency>
     <dependency>
       <groupId>com.azure</groupId>
-      <artifactId>azure-storage-common</artifactId>
-      <version>12.11.0</version> <!-- {x-version-update;com.azure:azure-storage-common;dependency} -->
+      <artifactId>azure-core-http-netty</artifactId>
+      <version>1.9.2</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/pom.xml:48]: 👍

**Review by srnagar — APPROVED** [review]

**Pull request merged**